File: exercises/ArraysAndStrings/1750-minimumLength/minimumLength.py
def minimumLength(s: str) -> int:
    left, right = 0, len(s) - 1

    while left < right and s[left] == s[right]:
        charToSkip = s[left]
        while left <= right and s[left] == charToSkip:
            left += 1
        while right >= left and s[right] == charToSkip:
            right -= 1

    return right - left + 1


if __name__ == "__main__":
    testCases = ["ca", "cabaabac", "aabccabba"]
    for test in testCases:
        print(minimumLength(test))

File: exercises/ArraysAndStrings/Dailies/1750-minimumLength/minimumLength.cpp
#include <iostream>
#include <string>

int minimumLength(std::string s)
{
    int l = 0;
    int r = s.size() - 1;

    // Two pointers approach. test if same char in suffix and prefix.
    while ((l < r) && s[l] == s[r])
    {
        char c = s[l];
        while ((l <= r) && s[l] == c)
            l++;

        // remember char on the right is the same as the left
        while ((r >= l) && s[r] == c)
            r--;
    }
    return r - l + 1;
}

int main()
{
    std::vector<std::string> testCases = {"ca", "cabaabac", "aabccabba"};

    for (auto test : testCases)
    {
        std::cout << minimumLength(test) << '\n';
    }
}